tools/nut-usbinfo.pl

```diff
@@ -125,17 +125,8 @@ sub gen_usb_files
 	print $outputUPower '# This file was automatically generated by NUT:'."\n";
 	print $outputUPower '# https://github.com/networkupstools/nut/'."\n#\n";
 	print $outputUPower '# To keep up to date, monitor upstream NUT'."\n";
-	print $outputUPower '# https://github.com/networkupstools/nut/commits/master/scripts/upower/95-upower-hid.rules'."\n";
+	print $outputUPower '# https://github.com/networkupstools/nut/commits/master/scripts/upower/95-upower-hid.hwdb'."\n";
 	print $outputUPower "# or checkout the NUT repository and call 'tools/nut-usbinfo.pl'\n\n";
```

Comment:
It may also help to leave a persistent link to update the file "in-place":
```suggestion
	print $outputUPower "# fetch https://raw.githubusercontent.com/networkupstools/nut/master/scripts/upower/95-upower-hid.hwdb\n";
```
...or maybe not, since support of devices depends on the binary NUT code deployed on the system, and this list seems generated from C sources (not `data/drivers.list` like in some other cases).

Reply:
We could also move the generation into the UPower repository. That is, if grabbing `data/drivers.list` from the NUT repository is enough to generate it there.

I suppose the reason for it being this way is that the data is parsed from the C source code in the first place.

Reply:
Looking at the files more, I think the `drivers.list` is not suitable here, it is more about documenting (often post-factum) that certain drivers understand and manage certain devices - how-ever those get branded on the market, etc. 

In particular, it is not about only USB, and does not convey VID/PID info systemically (for some devices these IDs are listed among comments or names, but not exhaustively nor in a parsable manner).

Reply:
> ... support of devices depends on the binary NUT code deployed on the system ...

Not sure what you mean here. NUT does not need to be installed for UPower to show warnings and trigger an emergency shutdown eventually.

Reply:
Guess I gotta read up more on what UPower does and how... I suppose it can deal with standard USB HID devices for at least basic commands/readings/alerts served on standard paths; however the `USB_DEVICE` macro also identifies the other USB connected UPSes, like those with effectively (protocol-wise) a built-in serial-to-USB converter and usually one of Megatec Qx protocol variations.

Reply:
It implements standard USB HID for reading basic status information. As I understand it, the code in the perl script is filtering the information to only include USB HID devices.

Reply:
Oh, right, found it at last:
````
                        # UPower device entry (only for USB/HID devices!)
                        if ($vendor{$vendorId}{$productId}{"driver"} eq "usbhid-ups") ...
````